azerothcore/azerothcore-wotlk
# chore(Core/AllMapScript) Update structure

## Changes Proposed:

This PR proposes changes to:
- [x] Hooks

## Issues Addressed:

- Closes

## Tests Performed:

This PR has been:

## How to Test the Changes:


1. When updating the crossfaction pve module, I tried to use the OnDestroyInstance hook and noticed that it was not being called. When checking the rest of the hooks, and remembering that the structure had been updated, I rewrote the files to adapt them and managed to get the hook to work in the module.
2. Basically, to test the pull request, you should use the hook without this change, and then use the hook with the change. In the first case, it should not be called, and in the second, it should execute without problems.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
